src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -51,10 +50,22 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
             SyntaxGenerator generator = editor.Generator;
             IArrayCreationOperation arrayArgument = GetArrayCreationOperation(node, model, cancellationToken, out bool isInvoked);
-            INamedTypeSymbol containingType = model.GetEnclosingSymbol(node.SpanStart, cancellationToken)!.ContainingType;
+            ISymbol enclosingSymbol = model.GetEnclosingSymbol(node.SpanStart, cancellationToken)!;
+            INamedTypeSymbol containingType = enclosingSymbol.ContainingType;
+            HashSet<string> identifiers = new(containingType.MemberNames);
+            if (enclosingSymbol is IMethodSymbol method)
+            {
+                identifiers.AddRange(method.Parameters.Select(p => p.Name));
+            }
+
+            IMethodBodyOperation? containingMethod = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody);
+            if (containingMethod?.BlockBody is not null)
+            {
+                identifiers.AddRange(containingMethod.BlockBody.Locals.Select(l => l.Name));
+            }
```

Comment:
This feels like a good intermediate fix for the issue being fixed, though I do wish we had access to https://github.com/dotnet/roslyn/blob/main/src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/NameGenerator.cs, which is not a public API.

Reply:
What's stopping us from making it public or copying it?